x-pack/test/functional/services/ml/trained_models_table.ts

```diff
@@ -226,6 +239,256 @@ export function TrainedModelsTableProvider(
+        'mlTrainedModelsInferenceAdvancedSettingsConditionTextArea',
+        values.condition
+      );
+      await this.setTrainedModelsInferenceFlyoutCustomValue(
+        'mlTrainedModelsInferenceAdvancedSettingsTagInput',
+        values.tag
+      );
+      await this.trainedModelsInferenceFlyoutSaveChanges();
+
+      await this.assertTrainedModelsInferenceFlyoutPipelineConfigValues(
+        values.editedInferenceConfig,
+        values.editedFieldMap
+      );
+      await this.assertTrainedModelsInferenceFlyoutAdditionalSettings(values.condition, values.tag);
+    }
+
+    public async completeTrainedModelsInferenceFlyoutDetails(
+      expectedValues: {
+        name: string;
+        description: string;
+        targetField: string;
+      },
+      editDefaults: boolean = false
+    ) {
+      if (editDefaults) {
+        await this.setTrainedModelsInferenceFlyoutCustomDetails(expectedValues);
+      }
+      const name = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineNameInput',
+        'value'
+      );
+      expect(name).to.eql(expectedValues.name);
+      const description = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineDescriptionInput',
+        'value'
+      );
+      expect(description).to.eql(expectedValues.description);
+      const targetField = await testSubjects.getAttribute(
+        'mlTrainedModelsInferencePipelineTargetFieldInput',
+        'value'
+      );
+      expect(targetField).to.eql(expectedValues.targetField);
+      await this.deployModelsContinue('mlTrainedModelsInferencePipelineProcessorConfigStep');
+    }
+
+    public async trainedModelsInferenceOpenAdditionalSettings() {
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsAccordion');
+      const additionalSettingsAccordionButton = await testSubjects.find(
+        'mlTrainedModelsInferenceAdvancedSettingsAccordionButton'
+      );
+      await additionalSettingsAccordionButton.click();
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsConditionTextArea');
+      await testSubjects.existOrFail('mlTrainedModelsInferenceAdvancedSettingsTagInput');
+    }
+
+    public async completeTrainedModelsInferenceFlyoutPipelineConfig(
+      expectedValues: {
+        inferenceConfig: any;
+        editedInferenceConfig?: any;
+        fieldMap: any;
+        editedFieldMap?: any;
+      },
+      editDefaults: boolean = false
+    ) {
+      const { inferenceConfig, editedInferenceConfig, fieldMap, editedFieldMap } = expectedValues;
+      // Check all defaults
+      await this.assertTrainedModelsInferenceFlyoutPipelineConfigValues(inferenceConfig, fieldMap);
+
+      if (editDefaults) {
+        await this.setTrainedModelsInferenceFlyoutCustomPipelineConfig({
+          condition: "ctx?.network?.name == 'Guest'",
```

Comment:
should it be a part of input params?

Reply:
It's just example text to make sure it gets saved correctly.

Reply:
I don't know if it's necessary now - it's just example text used in one place.